annotator/annotator.py

```diff
@@ -284,29 +264,36 @@ def to_json(self):
 
         return json.dumps(docless_spans)
 
-    def group_spans_by_containing_span(self, other_tier):
-        for span, next_span in pairs(self.spans):
-            if span.end > next_span.start:
-                raise Exception("Spans must be sorted and non-overlapping")
-        for span, next_span in pairs(other_tier.spans):
-            if span.end > next_span.start:
-                raise Exception("Spans must be sorted and non-overlapping")
-        other_spans = iter(other_tier.spans)
-        other_span = next(other_spans, None)
+    def group_spans_by_containing_span(self, other_tier, allow_partial_containment=False):
+        """
+        Group spans in the other tier by the spans that contain them.
+        """
+        other_spans = other_tier.spans
+        other_spans_idx = 0
         for span in self.spans:
             span_group = []
-            while True:
-                if other_span is None:
-                    break
-                if span.overlaps(other_span):
-                    if span.contains(other_span):
-                        span_group.append(other_span)
-                    else:
-                        print span, other_span
-                        raise Exception("Partial overlaps are not allowed")
-                elif other_span.end > span.end:
+            # iterate over the other spans that come before this span.
+            while other_spans_idx < len(other_spans):
+                if allow_partial_containment:
+                    if other_spans[other_spans_idx].end > span.start:
+                        break
+                else:
```

Comment:
The second clause is nested so that it aligns with the parallel clause above for easier comparison.